src/main/java/com/aws/greengrass/componentmanager/ComponentServiceHelper.java

```diff
@@ -110,6 +112,11 @@ ResolvedComponentVersion resolveComponentVersion(String componentName, Semver lo
                         .log(e.getMessage(), e);
                 throw new NoAvailableComponentVersionException(e.getMessage(), componentName, versionRequirements);
             }
+        } catch (GreengrassV2DataException e) {
+            if (RetryUtils.retryErrorCodes(e.statusCode())) {
+                throw new RetryableGGDeploymentException("Retrying on 500 error: " + e.statusCode());
```

Comment:
don't say it is a 500, you don't know that it actually is a 500. No message is needed, it won't be useful. Simply throw the exception `new RetryableDeploymentException(e)`